### PR TITLE
Replace Phabricator client with GitHub client.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
We want every pull request build to make the same reports as Phabricator
builds do.

Test Plan: manually

Subscribers: jenkins, marathon-team

JIRA issues: MARATHON-7602

### CODE REVIEW

File: ci/githubClient.sc [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    |request with the following changes in [buildinfo.json](https://github.com/dcos/dcos/blob/master/packages/marathon/buildinfo.json):
+    |
+    |$buildinfoDiff
+    |
+    |""".stripMargin
+
+  if (!hasUnsoundTests) {
+    msg += "**＼\\ ٩( ᐛ )و /／**"
+  } else {
+    val unsoundTestsList: String = unsoundTests.foldLeft("") { (msg:String, test: Js.Obj) =>
```
Comment (jeschkies) [inline ci/githubClient.sc]: Somehow this is not working currently even though the tests report a skipped test. We are not using the `KnownIssue` flag anyways. So I'd say we go forward without this extra comment.

File: ci/githubClient.sc [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    |
+    |**¯\\_(ツ)_/¯**
+    |""".stripMargin
+  }
+
+  comment(pullNumber, msg, event="APPROVE")
+}
+
+/**
+ * Report failue of diff build back to Phabricator.
```
Comment (kensipe) [inline ci/githubClient.sc]: failue -> failure 
Phabricator -> GH

File: ci/githubClient.sc [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import scalaj.http._
+import upickle._
+
+def execute(path:String, body: String): Unit = {
+  val GITHUB_API_TOKEN =
+    sys.env.getOrElse("GIT_PASSWORD", throw new IllegalArgumentException("GIT_PASSWORD enviroment variable was not set."))
+  val GITHUB_API_USER =
+    sys.env.getOrElse("GIT_USER", throw new IllegalArgumentException("GIT_USER enviroment variable was not set."))
+
+  // Execute request
```
Comment (kensipe) [inline ci/githubClient.sc]: I'm not sure the value of this comment

File: ci/githubClient.sc [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+ * Collect test results.
+ *
+ * @return the parsed test results.
+ */
+@main
+def collectTestResults(): Js.Arr = {
+
+  try {
+    // Join all results
+    val testResults = ls! pwd / 'target / "phabricator-test-reports" |? ( _.ext == "json")
```
Comment (kensipe) [inline ci/githubClient.sc]: "phabricator-test-reports" ?

File: ci/githubClient.sc [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    val joinedTestResults: Js.Arr = testResults.view.map(read!)
+      .map(upickle.json.read)
+      .collect { case a: Js.Arr => a }
+      .reduce { (l: Js.Arr, r: Js.Arr) =>
```
Comment (meln1k) [inline ci/githubClient.sc]: How about using `.fold` to handle a case when we don't have any results?

File: ci/githubClient.sc [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+import $file.utils
+import $file.awsClient
+
+import scala.util.control.NonFatal
+import scalaj.http._
+import upickle._
+
+def execute(path:String, body: String): Unit = {
+  val GITHUB_API_TOKEN =
+    sys.env.getOrElse("GIT_PASSWORD", throw new IllegalArgumentException("GIT_PASSWORD enviroment variable was not set."))
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/githubClient.sc]: Yes, it's not. :). Do you want it to be one?

File: ci/githubClient.sc [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+import ammonite.ops.ImplicitWd._
+
+import $file.utils
+import $file.awsClient
+
+import scala.util.control.NonFatal
+import scalaj.http._
+import upickle._
+
+def execute(path:String, body: String): Unit = {
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply ci/githubClient.sc]: Done.

File: ci/githubClient.sc (hunk 3 above)
Comment (jeschkies) [reply ci/githubClient.sc]: I've removed it.

File: ci/githubClient.sc (hunk 4 above)
Comment (jeschkies) [reply ci/githubClient.sc]: I did not want to take the time to read out the JUnit XML files. Ideally we have a different report that just collects the skipped tests. This reporter would then replace https://github.com/mesosphere/marathon/blob/master/project/PhabricatorTestReportListener.scala. It should be a follow up item.

File: ci/githubClient.sc (hunk 2 above)
Comment (jeschkies) [reply ci/githubClient.sc]: Done.

File: ci/githubClient.sc (hunk 5 above)
Comment (jeschkies) [reply ci/githubClient.sc]: Not sure. Don't we get an empty result anyways? Or do you mean if the files don't exist?

File: ci/githubClient.sc (hunk 4 above)
Comment (kensipe) [reply ci/githubClient.sc]: https://jira.mesosphere.com/browse/MARATHON-7780

File: ci/githubClient.sc (hunk 1 above)
Comment (kensipe) [reply ci/githubClient.sc]: https://jira.mesosphere.com/browse/MARATHON-7781

**Pull request merged**